### PR TITLE
DAOS-1988 vos: Port remaining vos txns to umem_tx

### PR DESCRIPTION
**Pull request by jolivier23**

### CODE REVIEW

File: src/bio/smd/smd_stream.c [hunk 1]
```diff
@@ -184,21 +185,20 @@ smd_nvme_add_stream_bond(struct smd_nvme_stream_bond *bond)
...
-		rc = dbtree_update(store->sms_stream_tab, &key, &value);
-		if (rc) {
-			D_ERROR("Adding a device : %d\n", rc);
-			pmemobj_tx_abort(ENOMEM);
-		}
-	} TX_ONABORT {
-		rc = umem_tx_errno(rc);
-		D_ERROR("Adding a stream bond entry: %d\n", rc);
-	} TX_END;
+	daos_iov_set(&key, &bond->nsm_stream_id, sizeof(int));
```
Comment (ryon-jensen) [inline src/bio/smd/smd_stream.c]: sizeof(int) makes me a little nervous  ... can it be sizeof(bond->nsm_stream_id) instead??

File: utils/build.config [hunk 2]
```diff
@@ -4,7 +4,7 @@ component=daos
 [commit_versions]
 ARGOBOTS = 31703b1e97b4aa9a8b2bfc18415ce5146e853c2a
 CART = 24c97eab7b97df49e8e26a4618157a806e92cbad
-PMDK = f9c64502747534a958326b72d694cf952c74c0c3
+PMDK = 1.5.1
```
Comment (ryon-jensen) [inline utils/build.config]: Using version numbers now instead of hash??

Comment (jolivier23) [reply utils/build.config]: You can specify a tag or a hash.  In this case, it’s a tagged release.  It corresponds to a hash

File: src/bio/smd/smd_stream.c (hunk 1 above)
Comment (jolivier23) [reply src/bio/smd/smd_stream.c]: Good point.  I didn’t change actually change it but you’re not wrong.

File: src/bio/smd/smd_internal.h [hunk 3]
```diff
@@ -171,6 +171,21 @@ smd_store_ptr2pop(struct smd_store *sms_obj)
 	return sms_obj->sms_uma.uma_pool;
 }
 
+static inline int
+smd_tx_begin(struct smd_store *sms_obj)
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/bio/smd/smd_internal.h]: > I don't think these wrapper functions have much value, and reader has to find the definition.

yeah, I it was the fact that I kept doing the same thing on commit/abort that caused me to add them initially and here I'm just repeating that.  Perhaps we should just add a umem_tx_end

**Pull request merged**